### PR TITLE
feat: add GlobalStyles via theme

### PR DESCRIPTION
**Pull request by viktoria-schwarz**
#1197

### CODE REVIEW

**Comment by viktoria-schwarz** [comment]
@sebald with a wrapping `<Box>` I could make it work. Or should I try to use emotion's `Global`? I tried to use it but couldn't access the theme (theme.styles.root) in a proper way... what do you think?

**Comment by sebald** [comment]
I would use `<Global>` since it does some extra stuff and optimization.

But I works not expose anything of that and keep it private to the `<MarigoldProvider>`.

And one more thing: since we don't do the `styles` thing, this is rather a theme-ui thing, use put "root" on the root theme object? 😄

**Comment by viktoria-schwarz** [comment]
In the `<body>` the styles work but not in the `<html>` when injected via the root in the theme. Not sure if I'm missing something :thinking:

File: packages/components/src/Provider/MarigoldProvider.tsx [hunk 1]
```diff
@@ -1,6 +1,32 @@
...
-import { ThemeProvider, ThemeProviderProps } from '@marigold/system';
+import {
+  ThemeProvider,
+  ThemeProviderProps,
+} from '@marigold/system';
+import { Global, jsx } from '@emotion/react';
+import { css, Theme } from '@theme-ui/css';
+
+const GlobalStyles = () =>
+  jsx(Global, {
```
Comment (sebald) [inline packages/components/src/Provider/MarigoldProvider.tsx:11]: You don't need the helper here. Just use JSX! -> `<Global>`

File: packages/components/src/Provider/MarigoldProvider.tsx [hunk 2]
```diff
@@ -1,6 +1,32 @@
...
+    styles: emotionTheme => {
+      const theme = emotionTheme as Theme;
+      const { useRootStyles } = theme.config || theme;
```
Comment (sebald) [inline packages/components/src/Provider/MarigoldProvider.tsx:14]: Do we need this?

File: packages/components/src/Provider/MarigoldProvider.tsx [hunk 3]
```diff
@@ -1,6 +1,32 @@
...
+
+      if (useRootStyles === false || (theme.styles && !theme.styles.root)) {
```
Comment (sebald) [inline packages/components/src/Provider/MarigoldProvider.tsx:16]: Ehem 🤪

**Comment by sebald** [comment]
@viktoria-schwarz I just took a quick look, but I think you copy/pasted something out shouldn't have 🤪

**Review by ti10le — APPROVED** [review]

**Pull request merged**